lightning/src/ln/channelmanager.rs

```diff
@@ -3755,7 +3755,8 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 						self.best_block.read().unwrap().height()));
 				self.fail_htlc_backwards_internal(channel_state.take().unwrap(),
 						HTLCSource::PreviousHopData(htlc.prev_hop), payment_hash,
-						HTLCFailReason::Reason { failure_code: 0x4000 | 15, data: htlc_msat_height_data });
+						HTLCFailReason::Reason { failure_code: 0x4000 | 15, data: htlc_msat_height_data },
+						self.get_our_node_id());
```

Comment:
I'm a bit confused why we're ever passing our own node id here - unless we're the ultimate recipient of an HTLC (in which case we probably shouldn't be generating a *forwarding* failure event) the recipient of the HTLC from our point of view is always some node we have a channel with.